src/main/java/org/candlepin/subscriptions/product/OfferingSyncController.java

```diff
@@ -129,11 +133,11 @@ private SyncResult syncOffering(Offering newState) {
     }
 
     // Update to the new entry or create it.
-    offeringRepository.save(newState);
+    offeringRepository.saveAndFlush(newState);
```

Comment:
Why are we switching to a save and flush here?

Reply:
I was still new with Spring Data at the time this was written, I did not realize that `save` did not necessarily mean that it gets committed to the DB. At this point in the sync call, it is expected that the offering should be updated within the DB.

If my understanding of `save` and `saveAndFlush` are not correct, I can switch it back.

Reply:
It may be worth mentioning this isn't wrapped in a `@Transactional` annotation, which I figured meant that I needed to flush it manually like this rather than rely on the auto-magicks that happen in such cases.

Reply:
Got it, makes sense to me!